File: code_snippets/ts/src/develop/state.ts
import * as restate from "@restatedev/restate-sdk";

export const router = restate.object({
  name: "State",
  handlers: {
    greet: async (ctx: restate.ObjectContext, name: string) => {
      // <start_statekeys>
      const stateKeys = ctx.stateKeys();
      // <end_statekeys>

      // <start_get>
      const myString = (await ctx.get<string>("my-string-key")) ?? "my-default";
      const myNumber = (await ctx.get<number>("my-number-key")) ?? 0;
      // <end_get>

      // <start_set>
      ctx.set("my-key", "my-new-value");
      // <end_set>

      // <start_clear>
      ctx.clear("my-key");
      // <end_clear>

      // <start_clear_all>
      ctx.clearAll();
      // <end_clear_all>
    },
  },
});

File: code_snippets/go/develop/state.go
package develop

import (
	restate "github.com/restatedev/sdk-go"
)

type State struct{}

func (State) Greet(ctx restate.ObjectContext, greeting string) error {
	// <start_statekeys>
	stateKeys, err := restate.Keys(ctx)
	if err != nil {
		return err
	}
	// <end_statekeys>
	_ = stateKeys

	// <start_get>
	myString := "my-default"
	if s, err := restate.Get[*string](ctx, "my-string-key"); err != nil {
		return err
	} else if s != nil {
		myString = *s
	}

	myNumber, err := restate.Get[int](ctx, "my-number-key")
	if err != nil {
		return err
	}
	// <end_get>
	_ = myString
	_ = myNumber

	// <start_set>
	restate.Set(ctx, "my-key", "my-new-value")
	// <end_set>

	// <start_clear>
	restate.Clear(ctx, "my-key")
	// <end_clear>

	// <start_clear_all>
	restate.ClearAll(ctx)
	// <end_clear_all>

	return nil
}
